src/main/java/com/github/pages/AbstractPage.java

```diff
@@ -5,19 +5,25 @@
 
 public abstract class AbstractPage {
 
-    protected abstract String url();
+    private final static String ERROR_MESSAGE_SELECTOR = "//*[@role='alert']";
```

Comment:
I think it makes sense to create BaseHerokuPage and move this locator and getMessage() method there since AbstractPage shouldn't have any dependencies on the testable application(s)